ui/plugin-system/src/components/TimeSeriesQueryEditor.tsx

```diff
@@ -11,24 +11,137 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { Box, BoxProps } from '@mui/material';
-import { TimeSeriesQueryDefinition } from '@perses-dev/core';
+import { useState } from 'react';
 import { produce } from 'immer';
+import { Box, BoxProps, Button, IconButton, Stack, Typography } from '@mui/material';
+import AddIcon from 'mdi-material-ui/Plus';
+import DeleteIcon from 'mdi-material-ui/DeleteOutline';
+import ChevronDown from 'mdi-material-ui/ChevronDown';
+import ChevronRight from 'mdi-material-ui/ChevronRight';
+import { TimeSeriesQueryDefinition, QueryDefinition } from '@perses-dev/core';
+import { usePlugin, usePluginRegistry } from '../runtime';
 import { PluginEditor, PluginEditorProps } from './PluginEditor';
 
+const DEFAULT_QUERY_PLUGIN_TYPE = 'TimeSeriesQuery';
+
+export interface TimeSeriesQueryEditorProps {
+  queries: TimeSeriesQueryDefinition[];
+  onChange: (queries: QueryDefinition[]) => void;
+}
+
+export function TimeSeriesQueryEditor(props: TimeSeriesQueryEditorProps) {
+  const { onChange, queries } = props;
+  const hasMoreThanOneQuery = queries.length > 1;
+  const { defaultPluginKinds } = usePluginRegistry();
+  const defaultTimeSeriesQueryKind = defaultPluginKinds?.[DEFAULT_QUERY_PLUGIN_TYPE] ?? '';
+
+  const { data: defaultQueryPlugin } = usePlugin(DEFAULT_QUERY_PLUGIN_TYPE, defaultTimeSeriesQueryKind, {
+    useErrorBoundary: true,
+    enabled: true,
+  });
+
+  // State for which queries are collapsed
+  // TODO: Would be easier if we had IDs for queries.
```

Comment:
I could see it being nice to add identifiers at some point for a variety of use cases, but you're already doing so much in these refactors that it would be nice not to have to fix that too. 😄

Reply:
+1